### PR TITLE
fix: align video diffusion pipeline with TRT-LLM 1.3.0rc9 API changes

### PR DESCRIPTION
**Pull request by indrajit96**
#### Overview:

Aligns the Dynamo video diffusion pipeline with the TensorRT-LLM `1.3.0rc8` API changes and adds CI coverage to prevent future breakage during upgrades.

#### Details:

Two config fields became stale after the rc8 upgrade:

- torch_compile_mode (str) — dropped upstream; TRT-LLM now uses TorchCompileConfig.enable_torch_compile (bool) + the existing --disable-torch-compile flag
- warmup_steps (int) — replaced upstream by skip_warmup (bool) on VisualGenArgs

E2E Test

- Added `agg_video_diffusion.sh` launch script for aggregated video diffusion serving
- Added video_diffusion config to `test_trtllm.py` with VideoGenerationPayload for the /v1/videos endpoint
- Test uses `--skip-warmup`, `--disable-torch-compile`, and small resolution (480×272, 17 frames) to fit within L4 GPU memory
- Marked pre_merge with gpu_1

Container deps:

- Added imageio>=2.37.0 and imageio-ffmpeg>=0.6.0 to requirements.common.txt so video frame → MP4 encoding works out of the box

Manual Test

Curl command 

```
curl -s -X POST http://localhost:8000/v1/videos \
  -H "Content-Type: application/json" \
  -d '{
    "prompt": "Epic wide shot of climbers ascending Mount Everest, dramatic snow-covered peaks, cinematic lighting, high detail, 4K quality",
    "model": "Wan-AI/Wan2.1-T2V-1.3B-Diffusers",
    "seconds": 4,
    "size": "832x480",
    "nvext": {
      "fps": 24,
      "num_inference_steps": 50,
      "guidance_scale": 5.0,
      "negative_prompt": "blurry, low quality, distorted, deformed, ugly, pixelated",
      "seed": 42
    }
  }' | python3 -m json.tool
  ```
  
  Result : 
  
 https://github.com/user-attachments/assets/e4fa1982-e25a-4855-818d-9dd9d035ee4a

#### Where should the reviewer start?

- `components/src/dynamo/trtllm/engines/diffusion_engine.py` — the key one-line fix forwarding skip_warmup to PipelineLoader.load()
- `components/src/dynamo/trtllm/configs/diffusion_config.py` — config field changes
- `tests/serve/test_trtllm.py` — new E2E test and VideoGenerationPayload

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)

- closes GitHub issue: #xxx

### CODE REVIEW

File: components/src/dynamo/trtllm/engines/diffusion_engine.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ async def initialize(self) -> None:
         # Use PipelineLoader for the full loading flow:
         #   VisualGenArgs → DiffusionModelConfig → AutoPipeline → BasePipeline
         loader = PipelineLoader(diffusion_args)
-        self._pipeline = loader.load()
+        self._pipeline = loader.load(skip_warmup=self.config.skip_warmup)
```
[the remark this answers is not in the document]
Comment (indrajit96) [reply components/src/dynamo/trtllm/engines/diffusion_engine.py:127]: The skip_warmup on VisualGenArgs (line 170) and the skip_warmup on loader.load() (line 127) are not redundant — they serve different purposes.
PipelineLoader.load() uses its own skip_warmup parameter to decide whether to call pipeline.warmup(). It does not read self.args.skip_warmup.

File: tests/serve/test_trtllm.py [hunk 2]
```diff
@@ -24,10 +25,40 @@
...
     multimodal_payload_default,
 )
+from tests.utils.payloads import BasePayload
 
 logger = logging.getLogger(__name__)
 
 
+@dataclass
+class VideoGenerationPayload(BasePayload):
+    """Payload for /v1/videos endpoint (TRT-LLM video diffusion)."""
```
Comment (KrishnanPrash) [inline tests/serve/test_trtllm.py:35]: Not familiar with the current state of video support for vLLM and SGLang, but can we move this to Payload class to `tests/utils/payloads.py` for test_vllm/test_sglang.py to utilize it?

Comment (indrajit96) [reply tests/serve/test_trtllm.py:35]: Yes, good call, we can consolidate the generation payload classes into `tests/utils/payloads.py` so they're reusable across backends. The plan would be:

Move `VideoGenerationPayload`, `ImageGenerationPayload`, and `I2VPayload` from `test_trtllm.py` / `test_vllm_omni.py` into `tests/utils/payloads.py`
Add convenience builder functions (`video_generation_payload`, `image_generation_payload`) to `tests/utils/payload_builder.py`, following the existing pattern for `chat_payload`, `completion_payload`, etc.
Update `test_trtllm.py` and `test_vllm_omni.py` to import from the shared location
Let's keep this PR scoped to the trtllm version changes. 
I'll file a separate ticket for the payload consolidation.

**Review by KrishnanPrash — APPROVED** [review]

File: container/deps/requirements.common.txt [hunk 3]
```diff
@@ -7,6 +7,10 @@
 fastapi==0.120.1
 grpcio-tools<=1.76.0  # May have platform-specific builds; pins grpcio ecosystem version
 httpx==0.28.1
+
+# Video generation: encode frames to MP4 (used by TRT-LLM, vLLM-Omni, SGLang diffusion)
```
Comment (krishung5) [inline container/deps/requirements.common.txt:11]: Q: Do you know if we install these two deps for SGL and vLLM Omni somewhere else? With adding these deps into the `requirements.common.txt`, I think we could remove those installation in other places.

**Review by krishung5 — APPROVED** [review]

File: components/src/dynamo/trtllm/request_handlers/video_diffusion/video_handler.py [hunk 4]
```diff
@@ -239,8 +239,9 @@ async def generate(
 
             # Encode media based on what the pipeline returned
             if output.video is not None:
-                # Video output: torch.Tensor (num_frames, H, W, 3) uint8 → MP4
-                frames_np = output.video.cpu().numpy()
+                # MediaOutput.video is (B, T, H, W, C) uint8 since TRT-LLM rc9;
+                # squeeze the batch dim to get (T, H, W, C) for MP4 encoding.
+                frames_np = output.video[0].cpu().numpy()
```
Comment (nv-yna) [inline components/src/dynamo/trtllm/request_handlers/video_diffusion/video_handler.py]: could we have more defenses? e.g.
```
  video = output.video
  assert video.ndim == 5 and video.shape[0] == 1, (
      f"Expected video shape (1, T, H, W, C), got {video.shape}"
  )
  frames_np = video[0].cpu().numpy()
```

**Review by zhongdaor-nv — APPROVED** [review]

**Comment by zhongdaor-nv** [comment]
LGTM, please address other comments as well.

File: components/src/dynamo/trtllm/request_handlers/video_diffusion/video_handler.py (hunk 4 above)
Comment (nv-yna) [reply components/src/dynamo/trtllm/request_handlers/video_diffusion/video_handler.py]: after you add this, I think unit test will fail where components/src/dynamo/trtllm/tests/test_trtllm_video_diffusion.py line 542 need to be updated because the current mock uses (T, H, W, C) format

File: components/src/dynamo/trtllm/engines/diffusion_engine.py [hunk 5]
```diff
@@ -260,7 +260,7 @@ def generate(
 
         req = DiffusionRequest(
             request_id=0,
-            prompt=prompt,
+            prompt=[prompt],
```
Comment (nv-yna) [inline components/src/dynamo/trtllm/engines/diffusion_engine.py:263]: IIUC there is no test coverage that verifies the prompt format is a string, not a List[str]. It would be great to add test coverage to assess type

proposal from AI:
```
  def test_generate_wraps_prompt_as_list(self):
      """Verify DiffusionEngine passes prompt as List[str] to DiffusionRequest."""
      engine = DiffusionEngine(config=self.config, device="cpu")
      engine._initialized = True
      engine._pipeline = MagicMock()
      engine._pipeline.infer.return_value = SimpleNamespace(
          video=torch.zeros((1, 4, 64, 64, 3), dtype=torch.uint8),
          image=None, audio=None,
      )

      engine.generate(prompt="a golden retriever", height=64, width=64,
                      num_frames=4, num_inference_steps=1)

      req = engine._pipeline.infer.call_args[0][0]
      assert isinstance(req.prompt, list), f"Expected list, got
  {type(req.prompt)}"
      assert req.prompt == ["a golden retriever"]
```

**Review by nv-yna — APPROVED** [review]
Early approval from me. The PR is in a good state. LGTM after add more defenses / tests and resolve my comments.

File: container/deps/requirements.common.txt (hunk 3 above)
Comment (indrajit96) [reply container/deps/requirements.common.txt:11]: Good find! 
I checked and these two deps (`imageio`, `imageio-ffmpeg`) are not installed separately for SGLang or vLLM Omni elsewhere in the repo. 
`SGLang` and `vLLM Omni` just do runtime imports (with helpful error messages if missing). 
We need to add this to enable CI.
So I think we are good here, no removl needed.

File: components/src/dynamo/trtllm/request_handlers/video_diffusion/video_handler.py (hunk 4 above)
Comment (indrajit96) [reply components/src/dynamo/trtllm/request_handlers/video_diffusion/video_handler.py]: Good catch!
Fixed.

Just a side note test was still passing because
Test mocks used the old shape (4, 64, 64, 3) 4D, no batch dim. 
When the handler does [0] on that, it just returns (64, 64, 3) (a single frame). 
That's a valid numpy array, so `encode_to_mp4_bytes` happily encodes it and the test passes.

File: components/src/dynamo/trtllm/engines/diffusion_engine.py (hunk 5 above)
Comment (indrajit96) [reply components/src/dynamo/trtllm/engines/diffusion_engine.py:263]: Makes sense.
Added another test as per suggestion.
